gaia/geo/processes_raster.py

```diff
@@ -141,3 +144,100 @@ def compute(self):
                                      allBands=all_bands,
                                      output_type=otype,
                                      format=self.output_format)
+
+
+class ClusterProcess(GaiaProcess):
+    """
+    Cluster raster dataset to produce cluster assignment raster.
+    """
+    #: Tuple of required inputs; name, type , max # of each; None = no max
+    required_inputs = [
+        {'description': 'Raster to cluster',
+         'type': types.RASTER,
+         'max': 1
+         }
+    ]
+    #: Required arguments for the process
+    required_args = [
+        {
+            'name': 'method',
+            'title': 'Method',
+            'description': 'Clustering algorithm to use.',
+            'type': str
+        },
+        {
+            'name': 'k',
+            'title': 'k',
+            'description': 'Number of clusters.',
+            'type': int
+        }
+    ]
+
+    #: Default output format for the process
+    default_output = formats.RASTER
+
+    def __init__(self, method, k, **kwargs):
+        """
+        Create a process for raster cell clustering.
+
+        :param inputs: Raster to run clustering on.
+        :param method: Clustering algorithm to use.
+        :param k: Number of clusters to use.
+        :param kwargs: Other keyword arguments.
+        :return: ClusterProcess object.
+        """
+        self.method = method
+        self.k = k
+        super(ClusterProcess, self).__init__(**kwargs)
+        if not self.output:
+            self.output = RasterFileIO(name='result', uri=self.get_outpath())
+
+    def compute(self):
+        """
+        Cluster cells, create a raster cluster assignment dataset.
+        """
+
+        def feature_extraction(data):
+            """Handle feature extraction from 3D raster."""
+            # Reshape data as observations by features. Each band is a feature.
+            points = np.array([data[:, i, j]
```

Comment:
thanks @andrenguyen-bah so that I understand it completely, so for multi-band, we will update/modify this function perhaps and for now we assume to handle only one band? If that is the case can we document it and if we get multiband throw an error?